addons/account/models/account_order_line_mixin.py

```diff
@@ -0,0 +1,111 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+from odoo.tools import format_date
+
+
+class AccountOrderLineMixin(models.AbstractModel):
+    # This class enables down payment support for child records of account.order.mixin that can be down payments
+    _name = 'account.order.line.mixin'
+    _description = 'Account Order Line Mixin'
+
+    is_downpayment = fields.Boolean(string="Is a down payment",
+                                    help="Down payments are made when creating account moves from an order."
+                                         " They are not copied when duplicating an order.",
+                                    )
+
+    # To override
+    move_line_ids = fields.One2many('account.move.line')
+    display_type = fields.Selection(
+        [('line_section', "Section"), ('line_note', "Note")],
+        default=False,
+        help="Technical field for UX purpose.",
+    )
+    product_id = fields.Many2one('product.product')
+    name = fields.Text(compute='_compute_name')
+    order_id = fields.Many2one('account.order.mixin')
+    qty_to_invoice = fields.Float()
+    price_unit = fields.Float()
+    tax_ids = fields.Many2many('account.tax')
+    product_uom = fields.Many2one('uom.uom')
```

Comment:
Here as well, things should be factorized. No ondelete='restrict' in purchase ?

Reply:
product_uom_id btw ^^

Reply:
Also, shouldn't the ondelete='restrict' used in sales be kept in the generic mixin ? (it's not by default since the field is not required, I think ; its "set null"). If I try deleting a UOM that was used on some SO/PO, I expect the operation to fail, don't I ? ? :thinking: